docs/source/documentation/concepts.rst

```diff
@@ -177,3 +177,39 @@ There are two ways to run several tasks in parallel on a same organization. The
 .. TODO:: Detail vertical and horizontal scaling
 
 .. TODO:: Explain what is substra tools
+
+
+Compute plan execution
+-----------------------
+
+Read this paragraph to understand what happens during the compute plan execution and what can be done to improve the execution time.
+
+Once a compute plan is submitted to the platform, its tasks are scheduled to be executed on each organization.
+
+On each organization, Substra fetches the assets needed for the first task, builds the Docker image of the algo and creates a container with the relevant assets. The task executes and Substra saves its outputs.
+Afterwards, every task **from the same compute plan** that uses the same algo is executed in the same Docker container.
+
+Asset preparation
+^^^^^^^^^^^^^^^^^^
+
+The first step of the task execution is to fetch the necessary assets.
+These include the inputs (e.g. the algo or opener files), the output of other tasks (input artifacts of the task) and data samples.
+
+The assets, data samples excluded, come from the file systems of the organizations. If they are stored on other organizations, they are downloaded over HTTPS connections.
+Example: an algo submitted on another organization
+
+The data samples are stored on the organization, in a storage solution (MiniO). They are downloaded for the task, this may take a long time if the dataset is large.
+Example: depending on the deployment configuration, downloading hundreds of gigabytes may take a few hours.
+
+Since this step can be quite long, there is a cache system: on a given organization, all the downloaded files (assets and data samples) are saved on disk. So when another tasks reuses the same assets there is no need to download them again. One the cache is full, all its contents are deleted.
+
+Docker image build
+^^^^^^^^^^^^^^^^^^^
+
+For the first task of the compute plan that uses a given algo, Substra needs to build the Docker image and create the
+Docker container. This may take from a few minutes to an hour (or more), depending on the Docker image.
```

Comment:
An hour would be quite pessimistic, unless you install a lot of stuff in your container.

Reply:
It also depends on the size of the final image that need to be transfered to the private registry and then to the local kubernetes registry

Reply:
also I changed the previous sentence to "Substra needs to build the image, transfer it to the local registry then use it to spawn the container." 
I don't want to go into too much detail on the registry, as it seems to me like an implementation detail, and users are not familiar with registries